kubernetes/istio_details_service.go

```diff
@@ -147,16 +153,211 @@ func (in *IstioClient) GetDestinationRules(namespace string, serviceName string)
+// on a destination defined by namespace, serviceName and version parameters.
+// It returns false otherwise.
+func CheckDestinationPolicyCircuitBreaker(destinationPolicy IstioObject, namespace string, serviceName string, version string) bool {
+	if destinationPolicy == nil || destinationPolicy.GetSpec() == nil {
+		return false
+	}
+	_, hasCircuitBreaker := destinationPolicy.GetSpec()["circuitBreaker"]
+	if !hasCircuitBreaker {
+		return false
+	}
+	return filterByDestination(destinationPolicy.GetSpec(), namespace, serviceName, version)
+}
+
+// GetDestinationRulesSubsets returns an array of subset names where a specific version is defined for a given service
+func GetDestinationRulesSubsets(destinationRules []IstioObject, serviceName, version string) []string {
+	cfg := config.Get()
+	foundSubsets := make([]string, 0)
+	for _, destinationRule := range destinationRules {
+		if dName, ok := destinationRule.GetSpec()["name"]; ok && dName == serviceName {
+			if subsets, ok := destinationRule.GetSpec()["subsets"]; ok {
+				if dSubsets, ok := subsets.([]interface{}); ok {
+					for _, subset := range dSubsets {
+						if innerSubset, ok := subset.(map[string]interface{}); ok {
+							subsetName := innerSubset["name"]
+							if labels, ok := innerSubset["labels"]; ok {
+								if dLabels, ok := labels.(map[string]interface{}); ok {
+									if versionValue, ok := dLabels[cfg.VersionFilterLabelName]; ok && versionValue == version {
+										foundSubsets = append(foundSubsets, subsetName.(string))
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return foundSubsets
+}
+
+// CheckDestinationRuleCircuitBreaker returns true if the destinationRule object includes a trafficPolicy configuration
+// on connectionPool or outlierDetection.
+// TrafficPolicy configuration can be defined at service level or per subset defined by a version.
+// It returns false otherwise.
+func CheckDestinationRuleCircuitBreaker(destinationRule IstioObject, namespace string, serviceName string, version string) bool {
+	if destinationRule == nil || destinationRule.GetSpec() == nil {
+		return false
+	}
+	cfg := config.Get()
+	if dName, ok := destinationRule.GetSpec()["name"]; ok && dName == serviceName {
+		if trafficPolicy, ok := destinationRule.GetSpec()["trafficPolicy"]; ok && checkTrafficPolicy(trafficPolicy) {
+			return true
+		}
+		if subsets, ok := destinationRule.GetSpec()["subsets"]; ok {
+			if dSubsets, ok := subsets.([]interface{}); ok {
+				for _, subset := range dSubsets {
+					if innerSubset, ok := subset.(map[string]interface{}); ok {
+						if trafficPolicy, ok := innerSubset["trafficPolicy"]; ok && checkTrafficPolicy(trafficPolicy) {
+							if labels, ok := innerSubset["labels"]; ok {
+								if dLabels, ok := labels.(map[string]interface{}); ok && version != "" {
+									if dLabels[cfg.VersionFilterLabelName] == version {
```

Comment:
Should this use the same idiom:
` versionValue, ok := dLabels[cfg.VersionFilterLabelName]; ok && versionValue == version`